views/client.go

```diff
@@ -90,3 +191,141 @@ func startViewsQuery(playbackID, playbackRecordingID string) string {
 		queryID,
 	)
 }
+
+var viewershipBreakdownFields = map[string]string{
+	"device":      "device",
+	"os":          "os",
+	"user_agent":  "user_agent_family",
+	"continent":   "playback_continent_name",
+	"country":     "playback_country_name",
+	"subdivision": "playback_subdivisions_name",
+	"timezone":    "playback_timezone",
+	"viewer_id":   "viewer_id",
+}
+
+var allowedTimeSteps = map[string]bool{
+	"hour":  true,
+	"day":   true,
+	"week":  true,
+	"month": true,
+	"year":  true,
+}
+
+type LDSMStateRow struct {
```

Comment:
i'd renamed this. the table is called `viewership_events` something similar woudl make the most sense